Hardware/Hardware/CardReader/StripeCardReader/StripeCardReaderService.swift

```diff
@@ -281,13 +281,25 @@ extension StripeCardReaderService: CardReaderService {
                 case .success(let locationId):
                     return promise(.success(BluetoothConnectionConfiguration(locationId: locationId)))
                 case .failure(let error):
-                    let underlyingError = UnderlyingError(with: error)
+                    let underlyingError = self.underlyingError(from: error)
                     return promise(.failure(CardReaderServiceError.connection(underlyingError: underlyingError)))
                 }
             }
         }
     }
 
+    private func underlyingError(from rawError: LocalizedError) -> LocalizedError {
```

Comment:
I'm leaving this slightly confusingly named function (that doesn't necessarily make an `UnderlyingError`) because I think that the next steps are to make `UnderlyingError` an abstract parent of both `StripeUnderlyingError` (or something) and `CardReaderConfigError`

Reply:
I would prefer to stay away from creating a hierarchy of errors. All of them implement Error after all, so conforming to Error seems like is all we might need.

I do agree this function name could be less confusing. What about extracting it to a factory in UnderlyingError? Something like

```
extension UnderlyingError {
   init(with configurationError: CardReaderConfigurationError) {
   }
}
```

That way, I think, we could also avoid the downcast to CardReaderConfigError as well.

Reply:
Actually, I am not sure if it's actually possible to type the parameter as CardReaderConfigurationError, so the downcast might not be possible to avoid

Reply:
Yeah I tried to make the hierarchy and quickly noped out of that, so I'm glad you don't love that idea!

I'll work on the naming. I could maybe remove this method by using a failable initializer in the extension you suggest.